packages/nvidia_nat_mcp/src/nat/plugins/mcp/auth/service_account/provider_config.py

```
@@ -0,0 +1,137 @@
+        default_factory=dict,
+        description="Additional keyword arguments to pass to the custom function",
+    )
+
+    @model_validator(mode="after")
+    def validate_token_or_function(self):
+        """Ensure either token or function is provided, but not both."""
+        has_token = self.token is not None
+        has_function = self.function is not None
+
+        if not has_token and not has_function:
+            raise ValueError("Either 'token' or 'function' must be provided in service_token config")
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)